File: test/leetcode/m12/IntegerToRomanTest.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m12;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Test for {@link IntegerToRoman}
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/09/22 09:04
 */
class IntegerToRomanTest {
    private IntegerToRoman test;

    @BeforeEach
    void setUp() {
        test = new IntegerToRoman();
    }

    @Test
    void test1() {
        int num = 3749;
        String expected = "MMMDCCXLIX";
        assertEquals(expected, test.intToRoman(num));
    }

    @Test
    void test2() {
        int num = 58;
        String expected = "LVIII";
        assertEquals(expected, test.intToRoman(num));
    }

    @Test
    void test3() {
        int num = 1994;
        String expected = "MCMXCIV";
        assertEquals(expected, test.intToRoman(num));
    }
}

File: src/leetcode/m12/IntegerToRoman_test.go
package m12

import "testing"

func Test_intToRoman(t *testing.T) {
	tests := []struct {
		name string
		num  int
		want string
	}{
		{"Test case 1", 3749, "MMMDCCXLIX"},
		{"Test case 2", 58, "LVIII"},
		{"Test case 3", 1994, "MCMXCIV"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := intToRoman(tt.num); got != tt.want {
				t.Errorf("intToRoman() = %v, want %v", got, tt.want)
			}
		})
	}
}
